### PR TITLE
pcf8591: add ADC only implementation for I2C ADC/DAC

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds an ADC only implementation for the pcf8591 I2C ADC/DAC.

A future PR can add the DAC feature.

### CODE REVIEW

File: examples/pcf8591/main.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	"tinygo.org/x/drivers/pcf8591"
+)
+
+var (
+	i2c = machine.I2C0
+)
+
+func main() {
+	machine.I2C0.Configure(machine.I2CConfig{})
+	adc := pcf8591.New(machine.I2C0)
```
Comment (sago35) [inline examples/pcf8591/main.go]: It would be better to use i2c instead of machine.I2C0.

File: pcf8591/pcf8591.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	CH3     ADCPin
+}
+
+// ADCPin is the implementation of the ADConverter interface.
+type ADCPin struct {
+	machine.Pin
+	d *Device
+}
+
+// New returns a new MCP3008 driver. Pass in a fully configured SPI bus.
```
Comment (sago35) [inline pcf8591/pcf8591.go]: Please update it to the PCF8591 version

**Review by sago35 — CHANGES_REQUESTED** [review]
Except for the small parts I pointed out, it looked good to me

File: examples/pcf8591/main.go (hunk 1 above)
Comment (deadprogram) [reply examples/pcf8591/main.go]: Done.

File: pcf8591/pcf8591.go (hunk 2 above)
Comment (deadprogram) [reply pcf8591/pcf8591.go]: Thanks, that was copy/paste error on my part. :smile_cat:

**Comment by deadprogram** [comment]
Thanks for the quick feedback @sago35 changes made to this PR, please take a look.

File: pcf8591/pcf8591.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	CH3     ADCPin
+}
+
+// ADCPin is the implementation of the ADConverter interface.
+type ADCPin struct {
+	machine.Pin
+	d *Device
+}
+
+// New returns a new PCF8591 driver. Pass in a fully configured SPI bus.
```
Comment (sago35) [inline pcf8591/pcf8591.go]:
```suggestion
// New returns a new PCF8591 driver. Pass in a fully configured I2C bus.
```

**Review by sago35 — CHANGES_REQUESTED** [review]
OK except for the following

File: pcf8591/pcf8591.go (hunk 3 above)
Comment (deadprogram) [reply pcf8591/pcf8591.go]: Oops, that was another one. Fixed!

**Comment by deadprogram** [comment]
Thanks for the proofreading @sago35 changes made and pushed.

**Review by sago35 — APPROVED** [review]
LGTM

**Pull request merged**